### PR TITLE
Move Matrix test out of test_matrices.py

### LINKED ISSUE #18666: Move matrix tests out of matrices.py and common.py
I guess @Pristine-Cat had not finished the work for the part.
So I'd open up this issue to be grabbed up by more wider audiences.

I'd expect you to create each files below and migrate the tests from `test_matrices.py` and `test_common.py`. For example, you should migrate all the determinant-related tests to `test_determinant.py`.

- [x] `test_determinant.py`
- [x] `test_decompositions.py`
- [x] `test_eigen.py`
- [x] `test_reductions.py`
- [x] `test_subspaces.py`
- [x] `test_solvers.py`

### PR DESCRIPTION
**Pull request by Arpan612**
Moved Matrix tests out of test_matrices.py
#### References to other Issues or PRs

Partially Fixes #18666

#### Brief description of what is fixed or changed

The functions of Matrix tests were moved out of test_matrices.py to test_reductions.py
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/matrices/tests/test_reductions.py [hunk 1]
```diff
@@ -275,3 +280,70 @@ def test_rref():
             [1, 0, sqrt(x)*(-x + 1)/(-x**Rational(5, 2) + x),
                 0, 1, 1/(sqrt(x) + x + 1)]):
         assert simplify(i - j).is_zero
+
+def test_rank():
+    from sympy.abc import x
+    m = Matrix([[1, 2], [x, 1 - 1/x]])
```
Comment (Smit-create) [inline sympy/matrices/tests/test_reductions.py]: No need to import this here, already being imported at the top of file.

File: sympy/matrices/tests/test_reductions.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
 from sympy.matrices.common import _MinimalMatrix, _CastableMatrix
-from sympy.matrices.matrices import MatrixReductions
+from sympy.matrices.matrices import MatrixReductions, _find_reasonable_pivot_naive
 from sympy.testing.pytest import raises
-from sympy.matrices import Matrix
+from sympy.matrices import (
+    GramSchmidt, ImmutableMatrix, ImmutableSparseMatrix, Matrix,
+    SparseMatrix, casoratian, diag, eye, hessian,
+    matrix_multiply_elementwise, ones, randMatrix, rot_axis1, rot_axis2,
+    rot_axis3, wronskian, zeros, MutableDenseMatrix, ImmutableDenseMatrix, MatrixSymbol)
```
Comment (Smit-create) [inline sympy/matrices/tests/test_reductions.py]: Please import only the required functions/classes, it may lead to build failure because of unused imports

**Comment by Smit-create** [comment]
Also, you need to fix you release notes.` Partially fixes #____ `will automatically close the issue which is not what we want. You may write something like `Related to #____`

**Comment by Arpan612** [comment]
Can someone please help me...I am continuously stuck in this array_utils error?

File: sympy/codegen/array_utils.py [hunk 3]
```diff
@@ -1288,7 +1288,7 @@ def recognize_matrix_expression(expr):
     >>> recognize_matrix_expression(cg)
     Trace(A)
 
-    Recognize some more complex traces:
+    Recognize some more complex traces:        
     >>> expr = Sum(A[i, j]*B[j, i], (i, 0, N-1), (j, 0, N-1))
```
Comment (Smit-create) [inline sympy/codegen/array_utils.py]: Remove these trailing spaces at the end of this line

**Comment by Smit-create** [comment]
Please go through the review comments. Also please note to import only required items. To resolve the conflicts go through following steps:
git checkout master
git pull https://github.com/sympy/sympy/ master
git checkout iss18666
git merge master.

You will get a 'conflict' message which you need to resolve manually

**Comment by Smit-create** [comment]
Why #18945 is also addressing the same changes?. I suggest working on any one of them.

**Comment by smitgajjar** [comment]
Yes, otherwise merge conflicts might arise!

**Comment by sylee957** [comment]
I'll close this because #18945 looks like the same PR

**Pull request closed without merging**